src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs

```diff
@@ -46,9 +46,10 @@ public static SourceParameterSymbol Create(
             var name = identifier.ValueText;
             var location = new SourceLocation(identifier);
 
-            if (hasParamsModifier && parameterType.IsSZArray())
+            if (hasParamsModifier && parameterType.IsSZArray() && !identifier.IsKind(SyntaxKind.None))
```

Comment:
> && !identifier.IsKind(SyntaxKind.None) [](http://example.com/codeflow?start=62&length=39)

Can we simply ensure that ```hasParamsModifier``` is never true for an extension parameter? For example, `params` can be completely disallowed as a modifier on it. This way we won't need to special case various checks for extension parameters.